File: app/src/main/java/com/aurosaswatraj/countmycrunch/Chronometer/ChronometerActivity.java
package com.aurosaswatraj.countmycrunch.Chronometer;

import androidx.appcompat.app.AppCompatActivity;

import android.animation.ObjectAnimator;
import android.content.pm.ActivityInfo;
import android.graphics.Color;
import android.os.Bundle;
import android.os.SystemClock;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.Chronometer;
import android.widget.ImageView;

import com.aurosaswatraj.countmycrunch.Dialogs.UserDarkModeDialog;
import com.aurosaswatraj.countmycrunch.R;


public class ChronometerActivity extends AppCompatActivity {



   private  Button btnstart,btnstop;
    private ImageView icanchor;
    private Chronometer timeHere;

    private ObjectAnimator objectAnimator;
    UserDarkModeDialog darkModeDialog;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_chronometer);
        darkModeDialog=new UserDarkModeDialog();
        darkModeDialog.darkModeDialog(this,this);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT); // Fixed portrait orientation



        Window window = this.getWindow();
        window.setStatusBarColor(Color.parseColor("#2f3640"));

        btnstart=findViewById(R.id.btn_start);
        btnstop=findViewById(R.id.btn_stop);
        icanchor=findViewById(R.id.icanchor);
        timeHere=findViewById(R.id.timeHere);




        objectAnimator=ObjectAnimator.ofFloat(icanchor,"rotation",0f,360f);
        objectAnimator.setDuration(4000);
        objectAnimator.setRepeatCount(ObjectAnimator.INFINITE);
//        For rotation..! https://developer.android.com/guide/topics/graphics/prop-animation.html#views



//        create optional aniamtion
        btnstop.setAlpha(0);


        btnstart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
//                passing Animation
                objectAnimator.start();
//                icanchor.startAnimation(roundingalone);
                btnstop.animate().alpha(1).translationY(-80).setDuration(300).start();
                btnstart.animate().alpha(0).setDuration(300).start();
//                start time
                timeHere.setBase(SystemClock.elapsedRealtime());
                timeHere.start();
            }
        });

        btnstop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                objectAnimator.pause();
                btnstart.animate().alpha(1).translationY(-80).setDuration(300).start();
                btnstop.animate().alpha(0).setDuration(300).start();
                timeHere.stop();
            }
        });





    }
}

File: app/src/main/java/com/aurosaswatraj/countmycrunch/Chronometer/Chronometer_Activity.kt
package com.aurosaswatraj.countmycrunch.Chronometer

import android.animation.ObjectAnimator
import android.content.pm.ActivityInfo
import android.graphics.Color
import android.os.Bundle
import android.os.SystemClock
import androidx.appcompat.app.AppCompatActivity
import com.aurosaswatraj.countmycrunch.Dialogs.UserDarkModeDialog
import com.aurosaswatraj.countmycrunch.R
import kotlinx.android.synthetic.main.activity_chronometer2.*

class Chronometer_Activity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_chronometer2)
        val window = this.window
        window.statusBarColor = Color.parseColor("#2f3640")
        var darkModeDialog = UserDarkModeDialog()
        darkModeDialog.darkModeDialog(this, this)
        requestedOrientation =ActivityInfo.SCREEN_ORIENTATION_PORTRAIT // Fixed portrait orientation

        val counter = 0
        var objectAnimator = ObjectAnimator.ofFloat(i_canchor, "rotation", 0f, 360f)
        objectAnimator.setDuration(4000)
        objectAnimator.setRepeatCount(ObjectAnimator.INFINITE)

        btn_stop_.alpha= 0F

        btn_start_.setOnClickListener {

//                passing Animation
            objectAnimator.start()
//                icanchor.startAnimation(roundingalone);
            //                icanchor.startAnimation(roundingalone);
            btn_stop_.animate().alpha(1f).translationY(-80f).setDuration(300).start()
            btn_start_.animate().alpha(0f).setDuration(300).start()
//                start time
            //   start time
            time_Here.base = SystemClock.elapsedRealtime()
            time_Here.start()
        }

        btn_stop_.setOnClickListener {
            val elapsedtime: Long = SystemClock.elapsedRealtime() - time_Here.getBase()
            objectAnimator.pause()
            btn_start_.animate().alpha(1f).translationY(-80f).setDuration(300).start()
            btn_stop_.animate().alpha(0f).setDuration(300).start()
            time_Here.stop()
        }



    }
}